src/hints/execution.rs

```diff
@@ -885,6 +885,22 @@ pub fn cache_contract_storage(
     Ok(())
 }
 
+pub const ADD_RELOCATION_RULE: &str = "memory.add_relocation_rule(src_ptr=ids.src_ptr, dest_ptr=ids.dest_ptr)";
+
+pub fn add_relocation_rule(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let src_ptr = get_ptr_from_var_name(vars::ids::SRC_PTR, vm, ids_data, ap_tracking)?;
+    let dest_ptr = get_ptr_from_var_name(vars::ids::DEST_PTR, vm, ids_data, ap_tracking)?;
+    vm.add_relocation_rule(src_ptr, dest_ptr)?;
```

Comment:
Note that this will error if `src_ptr` is not a temporary. I'd insist on a test case, but that code path seems pretty obvious to me by now...

Reply:
Yeah I didn't implement tests for trivial hints like this one, I don't know how much we should worry about this type of situations. Let's figure it out later.